src/qibolab/instruments/qm/config/config.py

```diff
@@ -0,0 +1,125 @@
+        octave.RF_outputs[str(port)] = OctaveOutput.from_config(lo_config)
+        self.controllers[octave.connectivity].add_octave_output(port)
+
+        intermediate_frequency = config.frequency - lo_config.frequency
+        self.elements[channel.logical_channel.name] = RfOctaveElement(
+            channel.serial,
+            output_switch(octave.connectivity, channel.port),
+            intermediate_frequency,
+        )
+
+    def configure_acquire_line(
+        self,
+        acquire_channel: QmChannel,
+        probe_channel: QmChannel,
+        acquire_config: QmAcquisitionConfig,
+        probe_config: IqConfig,
+        lo_config: OscillatorConfig,
+    ):
+        port = acquire_channel.port
+        octave = self.octaves[acquire_channel.device]
+        octave.RF_inputs[str(port)] = OctaveInput(lo_config.frequency)
+        self.controllers[octave.connectivity].add_octave_input(port, acquire_config)
+
+        port = probe_channel.port
+        octave = self.octaves[probe_channel.device]
+        octave.RF_outputs[str(port)] = OctaveOutput.from_config(lo_config)
+        self.controllers[octave.connectivity].add_octave_output(port)
+
+        intermediate_frequency = probe_config.frequency - lo_config.frequency
+        self.elements[probe_channel.logical_channel.name] = AcquireOctaveElement(
+            probe_channel.serial,
+            acquire_channel.serial,
+            output_switch(octave.connectivity, probe_channel.port),
+            intermediate_frequency,
+            time_of_flight=acquire_config.delay,
+            smearing=acquire_config.smearing,
+        )
+
+    def register_iq_pulse(self, element: str, pulse: Pulse):
+        op = operation(pulse)
+        if op not in self.pulses:
+            self.pulses[op] = qmpulse = QmPulse.from_pulse(pulse)
+            waveforms = waveforms_from_pulse(pulse)
+            for mode in ["I", "Q"]:
+                self.waveforms[qmpulse.waveforms[mode]] = waveforms[mode]
+        self.elements[element].operations[op] = op
+        return op
+
+    def register_dc_pulse(self, element: str, pulse: Pulse):
+        op = operation(pulse)
+        if op not in self.pulses:
+            self.pulses[op] = qmpulse = QmPulse.from_dc_pulse(pulse)
+            self.waveforms[qmpulse.waveforms["I"]] = waveforms_from_pulse(pulse)["I"]
+        self.elements[element].operations[op] = op
+        return op
+
+    def register_acquisition_pulse(self, element: str, pulse: Pulse):
+        """Registers pulse, waveforms and integration weights in QM config."""
+        op = operation(pulse)
+        acquisition = f"{op}_{element}"
+        if acquisition not in self.pulses:
+            self.pulses[acquisition] = qmpulse = QmAcquisition.from_pulse(
+                pulse, element
+            )
+            waveforms = waveforms_from_pulse(pulse)
+            for mode in ["I", "Q"]:
+                self.waveforms[qmpulse.waveforms[mode]] = waveforms[mode]
```

Comment:
Same as above, with a toggle to switch from `QmPulse` to `QmAcquisition`.

Reply:
940dfc78fdaa65b0e52d651bc37db955872903a6